File: frontend/src/types/job.ts
export type VariableSpec = {
    name: string;
    lower: string;
    upper: string;
};

export type CreateJobRequest = {
    name: string;
    integrand: string;
    variables: VariableSpec[];
    evaluations: number;
};

export type CreateJobResponse = {
    jobId: string;
    status: string;
};

File: api-orchestrator/internal/job/job.go
package job

import "time"

type Status string

const (
	StatusQueued    Status = "queued"
	StatusRunning   Status = "running"
	StatusSucceeded Status = "succeeded"
	StatusFailed    Status = "failed"
	StatusCanceled  Status = "canceled"
)

type VariableSpec struct {
	Name  string `json:"name"`
	Lower string `json:"lower"`
	Upper string `json:"upper"`
}

type Job struct {
	ID                   string
	Name                 string
	Integrand            string
	IntegrationVariables []VariableSpec
	Evaluations          int
	Status               Status
	ErrorMessage         *string
	ResultJSON           *string
	CreatedAt            time.Time
	UpdatedAt            time.Time
}
